### PR TITLE
TextIt #3807

### PR DESCRIPTION
**Pull request by lcaresia**
To test the source will be needed integrate with Zapier inside Textit platform.

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4058"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/textit/sources/new-flow-event/new-flow-event.mjs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      return this.db.get("webhookId");
+    },
+    _setWebhookId(webhookId) {
+      this.db.set("webhookId", webhookId);
+    },
+    emitEvent(event) {
+      const data = event.data ?? event;
+
+      this.$emit(data, {
+        id: data.run.uuid,
```
Comment (feyzullah) [inline components/textit/sources/new-flow-event/new-flow-event.mjs:31]: Is it certain that `data.run` be defined? If not can we use `data.run?.uuid`?

File: components/textit/actions/start-flow/start-flow.mjs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "groupUuid",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const response = await this.textit.startFlow({
+      $,
+      data: {
+        flow: this.flowUuid,
+        contacts: [
```
Comment (feyzullah) [inline components/textit/actions/start-flow/start-flow.mjs:35]: How about changing `contactUuid` type to `string[]` and using it as `contacts:this.contactUuid,`?

File: components/textit/actions/start-flow/start-flow.mjs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          this.contactUuid,
+        ],
+        groups: [
```
Comment (feyzullah) [inline components/textit/actions/start-flow/start-flow.mjs:38]: Again how about changing `groupUuid` type to `string[]` and using it as `groups.groupUuid,`?

**Review by feyzullah — APPROVED** [review]
LGTM! Added just a few optional comments

**Pull request merged**